KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -28,15 +28,32 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     }
 
     func refreshToken(completion: @escaping (Result<Bool>) -> Void) {
-        self.callback = completion
+        callback = completion
 
         guard tokenNeedsRefreshing() else {
             completion(Result.success(result: false))
             return
         }
 
-        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken else {
-            completion(Result.failure(error: RefreshTokenError.noRefreshToken))
+        guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken, refreshToken.isEmpty == false else {
+            Karhoo.configuration.requestNewAuthenticationCredentials { [weak self] newCredentials in
+                guard let self = self else {
+                    completion(.failure(error: RefreshTokenError.memoryAllocationError))
+                    return
+                }
+
+                if let newCredentials = newCredentials {
+                    let newToken = AuthToken(
+                        accessToken: newCredentials.accessToken,
+                        expiresIn: Int(newCredentials.expiryDate?.timeIntervalSinceNow ?? 0),
+                        refreshToken: newCredentials.refreshToken ?? "",
+                        refreshExpiresIn: Int(newCredentials.refreshTokenExpiryDate?.timeIntervalSinceNow ?? 0)
```

Comment:
Are you sure that we should use 0 as default expiration interval in case we receive null from backend? Shouldn't we treat is as invalid data / error?

Reply:
Please bear in mind that in the scenario where there is no `refreshToken`, there won't be any `refreshExpiresIn` either. And that's ok. In that case it shouldn't be treated as an error and 0 or nil is ok as it will be ignored anyway. 

But if there is a valid `refreshToken` and no `refreshExpiresin` it may mean there's either no expiration date or there is a backend issue that needs addressing. Best to ask the partners team how to interpret this scenario before modifying this any further.